catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java

```diff
@@ -563,6 +595,31 @@ private void nfcInitialize() {
 		resourceInitialized();
 	}
 
+	private static ArrayList<Brick> getBricksRequieringResource(int resource) {
```

Comment:
Got that covered. 
Filtering could be done generic, but here we want to distinct the list by comparing one of their properties (the class) **to each other**. So not really filtering. Tried to implement it with `Lists.transform`, but does not seem right. 
Added a function to our own Utils-class, see in next push.